lib/gssapi/mech/gss_cred.c

```diff
@@ -99,6 +157,23 @@ gss_export_cred(OM_uint32 * minor_status,
 	_gss_secure_release_buffer(minor_status, &buffer);
     }
 
+    if (cred->gc_neg_mechs != GSS_C_NO_OID_SET) {
+	major = export_oid_set(minor_status, GSS_SPNEGO_MECHANISM,
+			       cred->gc_neg_mechs, &buffer);
+	if (major == GSS_S_COMPLETE) {
+	    bytes = krb5_storage_write(sp, buffer.value, buffer.length);
```

Comment:
Why not just pass the sp to export_oid_set()?  Why not just always `export_oid_set()` unconditionally?  If the OID set is empty or `GSS_C_NO_OID_SET` then `export_oid_set()` can just write a bit fat zero.

Reply:
Sure, I can pass `sp` to `export_oid_set()`; I was just trying to keep things close to the existing mechanism calls for simplicity (whilst I was understanding the code too). I'll fix that now. I don't think it's necessary to always export unconditionally though. Technically there's a difference between a null and empty OID set of neg mechs, although I don't know off hand how that plays out in the rest of the code...

Reply:
I expect we never see empty OID sets here.